flickit-assessment-kit/src/test/java/org/flickit/assessment/kit/application/service/answerrange/GetAnswerRangeListServiceTest.java

```diff
@@ -0,0 +1,126 @@
+package org.flickit.assessment.kit.application.service.answerrange;
+
+import org.flickit.assessment.common.application.domain.crud.PaginatedResponse;
+import org.flickit.assessment.common.exception.AccessDeniedException;
+import org.flickit.assessment.data.jpa.kit.answerrange.AnswerRangeJpaEntity;
+import org.flickit.assessment.kit.application.domain.AnswerRange;
+import org.flickit.assessment.kit.application.port.in.answerrange.GetAnswerRangeListUseCase;
+import org.flickit.assessment.kit.application.port.out.answerange.LoadAnswerRangePort;
+import org.flickit.assessment.kit.application.port.out.expertgroupaccess.CheckExpertGroupAccessPort;
+import org.flickit.assessment.kit.application.port.out.kitversion.LoadKitVersionPort;
+import org.flickit.assessment.kit.test.fixture.application.AnswerRangeMother;
+import org.flickit.assessment.kit.test.fixture.application.AssessmentKitMother;
+import org.flickit.assessment.kit.test.fixture.application.KitVersionMother;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Sort;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class GetAnswerRangeListServiceTest {
+
+    @InjectMocks
+    GetAnswerRangeListService service;
+
+    @Mock
+    LoadKitVersionPort loadKitVersionPort;
+
+    @Mock
+    CheckExpertGroupAccessPort checkExpertGroupAccessPort;
+
+    @Mock
+    LoadAnswerRangePort loadAnswerRangePort;
+
+    @Test
+    void testGetAnswerRangeListService_CurrentUserDoesNotHaveAccess_ThrowsAccessDeniedException() {
+        var param = createParam(GetAnswerRangeListUseCase.Param.ParamBuilder::build);
+        var kitVersion = KitVersionMother.createKitVersion(AssessmentKitMother.simpleKit());
+
+        when(loadKitVersionPort.load(param.getKitVersionId())).thenReturn(kitVersion);
+        when(checkExpertGroupAccessPort.checkIsMember(kitVersion.getKit().getExpertGroupId(), param.getCurrentUserId())).thenReturn(false);
+        assertThrows(AccessDeniedException.class, () -> service.getAnswerRangeList(param));
+    }
+
+    @Test
+    void testGetAnswerRangeListService_ValidParams_ReturnsPaginatedAnswerRangeWithOptions() {
+        int page = 0;
+        int size = 10;
+        var param = createParam(GetAnswerRangeListUseCase.Param.ParamBuilder::build);
+        var kitVersion = KitVersionMother.createKitVersion(AssessmentKitMother.simpleKit());
+        var answerRange1 = AnswerRangeMother.createAnswerRange();
+        var answerRange2 = AnswerRangeMother.createAnswerRange();
+
+        PaginatedResponse<AnswerRange> paginatedAnswerRanges = new PaginatedResponse<>(
+            List.of(answerRange1, answerRange2),
+            page,
```

Comment:
why not using `param.page()`?

Reply:
What a good question! I don't know :D.
Fixed.